mindsdb_native/libs/phases/data_cleaner/data_cleaner.py

```diff
@@ -1,30 +1,40 @@
 from mindsdb_native.libs.phases.base_module import BaseModule
-from mindsdb_native.libs.data_types.mindsdb_logger import log
 
 
 class DataCleaner(BaseModule):
-    def _cleanup_w_missing_targets(self, df):
-        initial_len = len(df)
-        df = df.dropna(subset=self.transaction.lmd['predict_columns'])
-        no_dropped = len(df) - initial_len
-        if no_dropped > 0:
-            self.log.warning(f'Dropped {no_dropped} rows because they had null values in one or more of the columns that we are trying to predict. Please always provide non-null values in the columns you want to predict !')
-        return df
-
-    def _cleanup_ignored(self, df):
+    def _get_empty_columns(self, df):
+        empty_columns = []
         for col_name in df.columns.values:
             if len(df[col_name].dropna()) < 1:
-                self.transaction.lmd['columns_to_ignore'].append(col_name)
-                self.transaction.lmd['empty_columns'].append(col_name)
+                empty_columns.append(col_name)
                 self.log.warning(f'Column "{col_name}" is empty ! We\'ll go ahead and ignore it, please make sure you gave mindsdb the correct data.')
 
-        try:
-            df = df.drop(columns=self.transaction.lmd['columns_to_ignore'])
-        except Exception as e:
-            pass
-        return df
+        return empty_columns
+
+    def _remove_missing_targets(self, df):
+        initial_len = len(df)
+        df.dropna(subset=self.transaction.lmd['predict_columns'], inplace=True)
+        no_dropped = initial_len - len(df)
+        if no_dropped > 0:
+            self.log.warning(
+                f'Dropped {no_dropped} rows because they had null values in one or more of the columns that we are trying to predict. Please always provide non-null values in the columns you want to predict !')
+
+    def _remove_duplicate_rows(self, df):
```

Comment:
Everything looks good, but this issue that I mentioned in the old deduplication PR still exists:
> Also, what if my data is thousands rows of random input to an AND gate? It will only keep 4 rows of data, so user should be `log.warned` about deduplication and must have an option to disable it when calling `Predictor.learn`.

Maybe we can at least add a flag in config `DEDUPLICATE_ROWS = True`?

Reply:
I agree with this @btseytlin please just add a `deduplicate` flag in the `advanced_arge`, set it to `True` by default but users can change it to `False`, besides that, please fix merge conflicts and go ahead and merge.